alf/algorithms/muzero_algorithm.py

```diff
@@ -343,6 +363,24 @@ def preprocess_experience(self, root_inputs: TimeStep,
             action = replay_buffer.get_field('prev_action', env_ids,
                                              positions[:, 1:])
 
+            observation = ()
+            if self._train_repr_prediction:
+                if type(self._data_transformer) != IdentityDataTransformer:
+                    observation = replay_buffer.get_field(
+                        'observation', env_ids, positions)
+                else:
+                    observation, step_type = replay_buffer.get_field(
+                        ('observation', 'step_type'), env_ids, positions)
+                    exp = alf.data_structures.make_experience(
+                        root_inputs, AlgStep(), state=())
+                    exp = exp._replace(
+                        step_type=step_type,
+                        observation=observation,
+                        batch_info=batch_info,
+                        replay_buffer=replay_buffer)
+                    exp = self._data_transformer.transform_experience(exp)
```

Comment:
I might mis-understood this part. 
But I'm curious why do we need to ``transform_experience`` when ``type(self._data_transformer)`` is  an ``IdentityDataTransformer``? 
We should ``transform_experience`` when it is not a ``IdentityDataTransformer``?

Reply:
Thanks for spotting it. This is introduced when I cleaned up for PR.